File: src/adaos/integrations/telegram/pairing.py
from __future__ import annotations
from typing import Mapping, Any, Optional


def extract_start_code(update: Mapping[str, Any]) -> Optional[str]:
    """Return code from '/start <code>' message if present, else None."""
    msg = (update.get("message") or {}) if isinstance(update, dict) else {}
    text = (msg.get("text") or "").strip()
    if not text:
        return None
    if text.lower().startswith("/start "):
        return text.split(" ", 1)[1].strip() or None
    return None

File: src/adaos/integrations/inimatic/backend/io/telegram/pairing.ts
// pairing/api.ts
export function extractStartCode(text: string | undefined | null): string | null {
	const t = (text || '').trim()
	if (!t) return null
	if (t.toLowerCase().startsWith('/start ')) {
		const code = t.split(' ', 2)[1]
		return (code || '').trim() || null
	}
	return null
}

